azureworker/types.py

```diff
@@ -0,0 +1,89 @@
+"""Implementation of 'azurefuncs' ABCs.
+
+Also includes internal facilities for marshaling
+and unmarshaling protobuf objects.
+"""
+
+
+import types
+import typing
+
+import azurefuncs
+
+from . import protos
+
+
+class HttpRequest(azurefuncs.HttpRequest):
+    """An HTTP request object.
+
+    The implemented API is a subset of Flask's Request:
+        http://flask.pocoo.org/docs/0.12/api/#incoming-request-data
+    """
+
+    __slots__ = ('_method', '_url', '_headers', '_params', '_body')
+
+    def __init__(self, method: str, url: str,
```

Comment:
We want to expose query string parameters here as well. (You likely already have this on the larger backlog - adding here as an fyi/reminder.)